From 26a383ba460ca20c11081428a64e37ae3489fd6e Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Fri, 4 Apr 2014 23:50:33 -0400
Subject: [PATCH] World.getNearbyEntities API

Adds much needed API for getting nearby entities based on a location and not an entity
---
 src/main/java/org/bukkit/World.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 80ca41a..a9cf99b 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -428,6 +428,10 @@ public interface World extends PluginMessageRecipient, Metadatable {
      *     match one or more of the given classes/interfaces
      */
     public Collection<Entity> getEntitiesByClasses(Class<?>... classes);
+    public Collection<Entity> getNearbyEntities(Location loc, int x, int y, int z); // EMC
+    public Collection<LivingEntity> getNearbyLivingEntities(Location loc, int x, int y, int z); // EMC
+    public Collection<Player> getNearbyPlayers(Location loc, int x, int y, int z); // EMC
+    public <T> Collection<T> getNearbyEntitiesByType(Class<? extends T> clazz, Location loc, int x, int y, int z); // EMC
 
     /**
      * Get a list of all players in this World
-- 
1.8.3.2

